### PR TITLE
insert and update on Crud Repository

### PR DESCRIPTION
**Pull request by njr-11**
CrudRepository was brought back into Jakarta Data, now inheriting from BasicRepository, with the insert, insertAll, update, and updateAll methods moved to there.  This pull gets us in sync with that and adds some tests.

### CODE REVIEW

**Comment by njr-11** [comment]
#build

File: dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java [hunk 1]
```diff
@@ -3438,6 +3438,75 @@ public void testRecordCrudRepositoryMethods() {
         assertEquals(0L, receipts.count());
     }
 
+    /**
+     * Tests all BasicRepository methods with a record as the entity.
```
Comment (mswatosh) [inline dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java]: This should be CRUDRepository, right?

**Review by mswatosh — DISMISSED** [review]

File: dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java (hunk 1 above)
Comment (njr-11) [reply dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java]: Yes, I'll get that corrected. Thanks for spotting it.

**Comment by njr-11** [comment]
#build
#libby

**Review by mswatosh — APPROVED** [review]

**Pull request merged**